### DISCUSSION TITLE
How to generate AI-on-the-edge-device__update__vxxxxxx.zip for local development?

### DISCUSSION THREAD

**Discussion by jasaw** [jomjol/AI-on-the-edge-device#3061] [Ideas] [answered]
I know how to compile and generate `firmware.bin` file that I OTA upgrade over wifi for development, but I can't seem to find any documentation on how to build the whole OTA upgrade zip file that includes firmware and html directory.

Can someone please give me some guidance on this? Thank you.

**Comment by SybexX** [comment]
github does it automatically https://github.com/jasaw/AI-on-the-edge-device/actions/runs/8951350224 or do you mean on the pc

**Comment by jasaw** [comment]
Yes, I mean generating the zip file on my development PC. This is what I understand so far:
1. Build the `firmware.bin`.
2. Generate the html tooltips: `cd tools/parameter-tooltip-generator && python generate-param-doc-tooltips.py`.
3. Create a zip file from command line? `zip -r test_update.zip html firmware.bin`

I uploaded this zip file to my esp32 but it complained the firmware file is corrupted. I removed the firmware file from the zip but it did not update any of the html file on the device. What am I doing wrong?

**Reply by SybexX** [reply]
I'm a Windows10 user and don't have much experience with Linux, so I can't help you here.
Take a look at https://github.com/jomjol/AI-on-the-edge-device/blob/rolling/.github/workflows/build.yaml, this is used by github to create the files, you may be able to you find something useful there.
I create a folder e.g. "AI-on-the-edge-device__update__rolling_(xxxxxxx)", I put the firmware.bin, the config and html folder in there and use 7-Zip to pack it.
![firmware_update](https://github.com/jomjol/AI-on-the-edge-device/assets/587201/9b5be855-226f-49d1-b982-4d8e43e982e5)

**Comment by jasaw** [comment]
@SybexX Thank you for pointing out the `build.yaml`. It is very helpful.

**Comment by jasaw** [comment, marked as answer]
For anyone who is interested, here's my script to package my development files into OTA upgrade zip file. It creates a `debug` directory and puts the zip file in there.

```sh
#!/bin/sh

sha_short=$(git rev-parse --short HEAD)


rm -rf ./html
mkdir html
cp -r ./sd-card/html/* ./html/
mkdir html/param-tooltips
cd tools/parameter-tooltip-generator
python generate-param-doc-tooltips.py
cd ../..

cp -r ./sd-card/html/* ./html/

echo "Replacing variables..."
cd html; find . -type f -exec sed -i "s/\$COMMIT_HASH/${sha_short}/g" {} \;
cd ..

rm -rf ./update
mkdir -p ./update
cp "./code/.pio/build/esp32cam/firmware.bin" "update/firmware.bin"
cp -r ./html ./update/
rm -rf ./update/config/
mkdir -p ./update/config/
cp ./sd-card/config/*.tfl ./update/config/ 2>/dev/null || true
cp ./sd-card/config/*.tflite ./update/config/ 2>/dev/null || true

rm -rf debug
mkdir -p debug
cd ./update
zip -r ../debug/AI-on-the-edge-device__update__debug.zip .

```
